src/app/icd/ICDManager.cpp

```diff
@@ -245,6 +245,11 @@ void ICDManager::UpdateICDMode()
     {
         ICDConfigurationData::GetInstance().SetICDMode(tempMode);
         postObserverEvent(ObserverEventType::ICDModeChange);
+
+        // Can't use attribute accessors/Attributes::FeatureMap::Get in unit tests
+#if !CONFIG_BUILD_FOR_HOST_UNIT_TEST
+        Attributes::OperatingMode::Set(kRootEndpointId, static_cast<OperatingModeEnum>(tempMode));
```

Comment:
now that we have an Enum from the datamodel for the Operating mode, we could change the type of ICDMode variable and this tempMode Variable and remove this cast

Reply:
We could but that implies either including /core in ICDConfigurationData or moving ICDMode out of ICDConfigurationData, which would be best left in a subsequent PR